docs/proposals/004-endpoint-picker-protocol/README.md

```diff
@@ -9,27 +9,53 @@ This doc defines the protocol between the EPP and the proxy (e.g, Envoy).
 The EPP MUST implement the Envoy
 [external processing service](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ext_proc/v3/external_processor) protocol.
 
+## Endpoint Subset
+For each HTTP request, the proxy CAN communicate the subset of endpoints the EPP MUST pick from by setting an unstructured entry in the [filter metadata](https://github.com/envoyproxy/go-control-plane/blob/63a55395d7a39a8d43dcc7acc3d05e4cae7eb7a2/envoy/config/core/v3/base.pb.go#L819) field of the ext-proc request. The metadata entry for the subset list MUST be wrapped with an outer key (which represents the metadata namespace) with a default of `envoy.lb.subset_hint`.
+
+```go
+filterMetadata: {
+  "envoy.lb.subset_hint" {
+     "x-gateway-destination-endpoint-subset-hint": [<ip:port>, <ip:port>, ...]
```

Comment:
how will this metadata be used, if this is set and returned to envoy, envoy must be configurred.

Reply:
The intention is for envoy/proxy to tell the EPP - "here is a list of endpoints that you can choose from". The EPP is likely to be expected to choose the best one **from that list**

Reply:
How to tell envoy to bring that info?